tests/device_tests/test_authenticate_device.py

```diff
@@ -46,48 +86,47 @@ def test_authenticate_device(session: Session, challenge: bytes) -> None:
 
     # Issue an AuthenticateDevice challenge to Trezor.
     proof = device.authenticate(session, challenge)
+
+    data = b"\x13AuthenticateDevice:" + compact_size(len(challenge)) + challenge
+
+    # Optiga
+
     certs = [x509.load_der_x509_certificate(cert) for cert in proof.optiga_certificates]
 
     # Verify the last certificate in the certificate chain against trust anchor.
     root_public_key = ec.EllipticCurvePublicKey.from_encoded_point(
-        ec.SECP256R1(), ROOT_PUBLIC_KEY[session.model]
+        ec.SECP256R1(), OPTIGA_ROOT_PUBLIC_KEY[session.model]
     )
     root_public_key.verify(
         certs[-1].signature,
         certs[-1].tbs_certificate_bytes,
         certs[-1].signature_algorithm_parameters,
     )
 
-    # Verify the certificate chain.
-    for cert, ca_cert in zip(certs, certs[1:]):
-        assert cert.issuer == ca_cert.subject
+    verify_cert_chain(certs, session.model.internal_name)
 
-        ca_basic_constraints = ca_cert.extensions.get_extension_for_class(
-            ext.BasicConstraints
-        ).value
-        assert ca_basic_constraints.ca is True
+    # Verify the signature of the challenge.
+    certs[0].public_key().verify(
+        proof.optiga_signature, data, ec.ECDSA(hashes.SHA256())
+    )
 
-        try:
-            basic_constraints = cert.extensions.get_extension_for_class(
-                ext.BasicConstraints
-            ).value
-            if basic_constraints.ca:
-                assert basic_constraints.path_length < ca_basic_constraints.path_length
-        except ext.ExtensionNotFound:
-            pass
+    # Tropic
 
-        ca_cert.public_key().verify(
-            cert.signature,
-            cert.tbs_certificate_bytes,
-            cert.signature_algorithm_parameters,
-        )
+    certs = [x509.load_der_x509_certificate(cert) for cert in proof.tropic_certificates]
 
-    # Verify that the common name matches the Trezor model.
-    common_name = cert.subject.get_attributes_for_oid(x509.oid.NameOID.COMMON_NAME)[0]
-    assert common_name.value.startswith(session.model.internal_name)
+    if session.model.internal_name in MODELS_WITHOUT_TROPIC:
+        pytest.xfail(f"{session.model.internal_name} does not support Tropic")
```

Comment:
Seem to me the test should rather succed in this case. But I'd rather split it in two testcases and use pytest.mark.models to determine which to run.